tensornetwork/block_sparse/block_tensor.py

```diff
@@ -1454,3 +1462,154 @@ def outerproduct(tensor1: BlockSparseTensor,
       flows=final_flows,
       order=tensor1._order + order2,
       check_consistency=False)
+
+
+def tensordot(tensor1: BlockSparseTensor,
+              tensor2: BlockSparseTensor,
+              axes: Optional[Union[Sequence[Sequence[int]], int]] = 2
+             ) -> BlockSparseTensor:
+  """
+  Contract two `BlockSparseTensor`s along `axes`.
+  Args:
+    tensor1: First tensor.
+    tensor2: Second tensor.
+    axes: The axes to contract.
+  Returns:
+      BlockSparseTensor: The result of the tensor contraction.
+  """
+
+  if isinstance(axes, (np.integer, int)):
+    axes = [
+        np.arange(tensor1.ndim - axes, tensor1.ndim, dtype=np.int16),
+        np.arange(0, axes, dtype=np.int16)
+    ]
+  elif isinstance(axes[0], (np.integer, int)):
+    axes = [np.array(axes, dtype=np.int16), np.array(axes, dtype=np.int16)]
+  axes1 = axes[0]
+  axes2 = axes[1]
+  if not np.all(np.unique(axes1) == np.sort(axes1)):
+    raise ValueError(
+        "Some values in axes[0] = {} appear more than once!".format(axes1))
+  if not np.all(np.unique(axes2) == np.sort(axes2)):
```

Comment:
Make sure all of these ValueError paths are tested so we can have 100% coverage.

Reply:
added more tests